contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -2987,3 +2994,22 @@ pltsql_planner_hook(Query *parse, const char *query_string, int cursorOptions, P
 
 	return plan;
 }
+
+static Node* 
+transform_like_in_add_constraint (Node* node)
+{
+	bool create_tbl_stmt_is_invalid = false;
+
+	if (current_query_is_create_tbl_check_constraint)
+		create_tbl_stmt_is_invalid = expr_contains_ilike_and_ci_collation_wrapper(node, true);
```

Comment:
Why do we need a wrapper? Can't we just call the callback function?

Reply:
I made a wrapper so we don't have to expose `init_and_check_collation_callbacks()`

Reply:
It is also being done for other collation callbacks (refer to [collation.h](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_2_X_DEV/contrib/babelfishpg_tsql/src/collation.h#L76))